### PR TITLE
fix(ios): v7 swipe to go back gestures with react navigation

### PR DESCRIPTION
**Pull request by intergalacticspacehighway**
# Summary
Resolves - https://github.com/callstack/react-native-pager-view/issues/709

- Adds swipe-to-go-back from the edge and `fullScreenGestureEnabled` support in scroll view implementation of pager view.
- This has a much cleaner approach thanks to the new scroll view based implementation, we don't need an extra pan gesture.

## Test Plan
Test example and fabric example with native stack and `fullScreenGestureEnabled`

### What's required for testing (prerequisites)?
Run `yarn install` and `pod install` in example and fabric example
### What are the steps to reproduce (after prerequisites)?
- Test if swipe to go back gesture from edge works
- Add `fullScreenGestureEnabled` and test if full screen swipe from the first page pops the screen.
## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ❌     |

## Checklist

- [x] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I updated the typed files (TS and Flow)

### CODE REVIEW

**Comment by intergalacticspacehighway** [comment]
open to changing the name of `RNCPageScrollView` since it's not an actual React Native Component. Couldn't decide so kept this for now 😅. Maybe just `PagerViewScrollView`?

File: ios/RNCPagerScrollView.m [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
+#import "RNCPagerScrollView.h"
+
+@implementation RNCPagerScrollView
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super initWithFrame:frame];
+    if (self) {
```
Comment (okwasniewski) [inline ios/RNCPagerScrollView.m:7]: Can you change this to 

```obj-c
if (self = [super initWithFrame:frame]) {
```

Just to have consistency across files.

File: ios/RNCPagerScrollView.m [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        self.panGestureRecognizer.delegate = self;
+    }
+    return self;
+}
+
+- (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldRecognizeSimultaneouslyWithGestureRecognizer:(UIGestureRecognizer *)otherGestureRecognizer {
```
Comment (okwasniewski) [inline ios/RNCPagerScrollView.m:13]: Also how about moving this delegate to an extension? It will make the code a bit cleaner.

**Review by okwasniewski — COMMENTED** [review]
Thanks for this! 🎉 I'm leaving two minor comments and I think we will be ready to merge.

File: ios/RNCPagerScrollView.m (hunk 2 above)
Comment (intergalacticspacehighway) [reply ios/RNCPagerScrollView.m:12]: I am a bit obj c noob. Do you mean to add a category and make a separate `RNCPagerScrollView+UIGestureRecognizerDelegate.m` file?

**Review by krozniata — APPROVED** [review]

File: ios/RNCPagerScrollView.m (hunk 2 above)
Comment (okwasniewski) [reply ios/RNCPagerScrollView.m:12]: Let's merge it as is since it's minimal change.

**Review by okwasniewski — APPROVED** [review]

**Pull request merged**